Sprint 22. Homework

DoNatPanic opened the pull request.

DoNatPanic committed:
Sprint 22. Homework. Part 1

DoNatPanic committed:
Sprint 22. Homework. Part 2

DoNatPanic committed:
Sprint 22. Homework. Part 3

kosstyango commented:
⚠️ Критические замечания:
Трек из Избранного не убирается при отжатии кнопки "Нравится" и не добавляется при нажатии на кнопку "Нравится"- обновление списка происходит только при перезаходе на фрагмент "Избранные треки" (см.видео-скрин).
https://github.com/user-attachments/assets/357daf05-7870-4a71-a40c-6263593d7175


Поломалась логика перехода в Аудиоплеер: первый раз из результатов поиска туда перейти можно, а вот второй и последующий - не работает (см.видео-скрин).
https://github.com/user-attachments/assets/1656b060-9f0e-4871-8880-d87e6da5a8d4

Приложение периодически падает при нажатии на кнопку "Медиатека" в боттом-шит меню (скрин ошибок)
![Спринт_22_Фаина3](https://github.com/user-attachments/assets/d95a35b0-1476-432c-ae40-213133b5f542)

🍏 Рекомендации, не влияющие на результат проверки работы:
История поиска НЕ показывается при очистке поискового запроса, а заголовок "Вы искали" иногда возникает неправильно - показывается над результатами предыдущего поиска БЕЗ связки с реальной историей поиска :(

DoNatPanic committed:
Fix critical problems

DoNatPanic commented:
Привет, Константин, и тебя с Новым годом!
Спасибо за ревью
Я поправила первые два замечания, насчет третьего - не смогла воспроизвести ошибку... а по логам на скрине я не разобрала в чем проблема возникла.

kosstyango commented:
теперь БАГ с Избранными треками отмечается в другой ситуации: При повторном просмотре трека в истории поиска, в нем не подсвечивается отметка "Нравится", хоть трек и записан в Избранное (см.видео-скрин)

https://github.com/user-attachments/assets/9edf8c5b-7460-40f0-8faa-86e4106d3fd2

DoNatPanic committed:
Fix critical problems 2

Tracks from tracks history is favourite  field update comparing with tracks from db

DoNatPanic commented:
Благодарю за найденный баг! Проблема была в том, что треки из истории запросов (которая хранится в виде текстового файла на устройстве) никак не сравнивались с треками из таблицы favourites_table в БД. Поэтому отметка "Нравится" всегда была в значении "нет".
Проблему решила.

DoNatPanic commented:
Благодаря видео-скрину сразу стало понятно что не так. Вопросов по исправлению не было, поэтому к помощи наставника не прибегала. 
Очень надеюсь, что третья попытка сдачи работы окажется удачной
